### PR TITLE
CppCheck Furnace

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes CppCheck suggestions
 - DESCRIBE PURPOSE OF THIS PULL REQUEST

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by rraustad** [comment]
Should probably wait on #9897 and #9940 to merge first. I can pull those changes into this branch.

File: src/EnergyPlus/DesiccantDehumidifiers.cc [hunk 1]
```diff
@@ -224,32 +224,31 @@ namespace DesiccantDehumidifiers {
...
+        Array1D<Real64> Numbers;        // Numeric input items for object
+        Array1D_bool lAlphaBlanks;      // Logical array, alpha field input BLANK = .TRUE.
+        Array1D_bool lNumericBlanks;    // Logical array, numeric field input BLANK = .TRUE.
+        bool errFlag;                   // local error flag
+        std::string RegenCoilType;      // Regen heating coil type
+        std::string RegenCoilName;      // Regen heating coil name
+        int SteamIndex;                 // steam coil Index
+        bool RegairHeatingCoilFlag;     // local error flag
 
         auto &MaxNums(state.dataDesiccantDehumidifiers->MaxNums);
```
Comment (amirroth) [inline src/EnergyPlus/DesiccantDehumidifiers.cc]: This is a bad use of a reference shortcut. Also, I am not sure why these need to be in `state` and can't just be loal variables.

Comment (rraustad) [reply src/EnergyPlus/DesiccantDehumidifiers.cc]: This is not used for anything.

image

Comment (rraustad) [reply src/EnergyPlus/DesiccantDehumidifiers.cc]: All 3 of these were in state ... were.

        int MaxNums = 0;
        int MaxAlphas = 0;
        int TotalArgs = 0;

File: src/EnergyPlus/DesiccantDehumidifiers.cc [hunk 2]
```diff
@@ -123,6 +123,7 @@ namespace DesiccantDehumidifiers {
     //              (Please see copyright and disclaimer information at end of module)
 
     static std::string const fluidNameSteam("STEAM");
+    Real64 constexpr TempSteamIn = 100.0;
```
Comment (rraustad) [inline src/EnergyPlus/DesiccantDehumidifiers.cc:126]: use and include DataGlobalConstants.hh: ??

    Real64 constexpr SteamInitConvTemp = 100.0;             // [deg C], standard init steam vol to mass flow conversion temp

**Comment by nrel-bot-2c** [comment]
@rraustad it has been 7 days since this pull request was last updated.

File: src/EnergyPlus/DataEnvironment.cc [hunk 3]
```diff
@@ -178,22 +176,17 @@ Real64 WindSpeedAt(EnergyPlusData &state, Real64 const Z) // Height above ground
     // 2005 ASHRAE Fundamentals, Chapter 16, Equation 4.  (Different depending on terrain).
     // Terrain variables are set in HeatBalanceManager or entered by the user.
 
-    // Return value
-    Real64 LocalWindSpeed; // Return result for function (m/s)
-
     if (Z <= 0.0) {
-        LocalWindSpeed = 0.0;
+        return 0.0;
```
Comment (Myoldmopar) [inline src/EnergyPlus/DataEnvironment.cc:180]: Love the early returns here.

File: src/EnergyPlus/DataZoneEquipment.cc [hunk 4]
```diff
@@ -942,34 +942,11 @@ void GetZoneEquipmentData(EnergyPlusData &state)
...
-    // PURPOSE OF THIS SUBROUTINE:
-    // Decide a few one-time things for later
-    // determination of flow regime for convection
-
-    // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-    int ZoneLoop;
-
-    for (ZoneLoop = 1; ZoneLoop <= state.dataGlobal->NumOfZones; ++ZoneLoop) {
-    }
-}
```
Comment (Myoldmopar) [inline src/EnergyPlus/DataZoneEquipment.cc:971]: Wow.

File: src/EnergyPlus/Furnaces.cc [hunk 5]
```diff
@@ -12636,52 +11385,52 @@ namespace Furnaces {
...
         // Set minimum OAT for heat pump compressor operation in heating mode
         errFlag = false;
-        if (furnace.HeatingCoilType_Num == Coil_HeatingAirToAirVariableSpeed) {
-            furnace.MinOATCompressorHeating = VariableSpeedCoils::GetVSCoilMinOATCompressor(state, furnace.HeatingCoilIndex, errFlag);
-        } else if (furnace.HeatingCoilType_Num == CoilDX_HeatingEmpirical) {
-            furnace.MinOATCompressorHeating = DXCoils::GetMinOATCompressor(state, furnace.HeatingCoilIndex, errFlag);
+        if (thisFurnace.HeatingCoilType_Num == DataHVACGlobals::Coil_HeatingAirToAirVariableSpeed) {
+            thisFurnace.MinOATCompressorHeating = VariableSpeedCoils::GetVSCoilMinOATCompressor(state, thisFurnace.HeatingCoilIndex, errFlag);
+        } else if (thisFurnace.HeatingCoilType_Num == DataHVACGlobals::CoilDX_HeatingEmpirical) {
+            thisFurnace.MinOATCompressorHeating = DXCoils::GetMinOATCompressor(state, thisFurnace.HeatingCoilIndex, errFlag);
```
Comment (Myoldmopar) [inline src/EnergyPlus/Furnaces.cc:11428]: Tons of good, typical cppcheck style, cleanups in here.

**Review by Myoldmopar — APPROVED** [review]
LGTM

**Comment by Myoldmopar** [comment]
Tested locally with develop pulled in, all happy.  Merging, thanks @rraustad

**Pull request merged**